substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/SubstrateJVM.java

```diff
@@ -140,6 +143,27 @@ public static JfrLogging getJfrLogging() {
         return get().jfrLogging;
     }
 
+    public static boolean setHandler(Class<? extends jdk.internal.event.Event> eventClass, EventHandler handler) {
+        try {
+            Field f = eventClass.getDeclaredField("eventHandler");
```

Comment:
Looking at the code path, this should never be reached at runtime since we expect `eventHandler` fields to already be in the image. I've removed this method and altered the substitution to throw an error.